ngraph/core/src/op/max_pool.cpp

```diff
@@ -223,3 +319,56 @@ shared_ptr<Node> op::v8::MaxPool::clone_with_new_inputs(const OutputVector& new_
                                     m_index_element_type,
                                     m_axis);
 }
+
+bool op::v8::MaxPool::has_evaluate() const {
+    NGRAPH_OP_SCOPE(v8_MaxPool_has_evaluate);
+    switch (get_input_element_type(0)) {
+    case ngraph::element::i8:
+    case ngraph::element::i32:
+    case ngraph::element::i64:
+    case ngraph::element::u8:
+    case ngraph::element::u32:
+    case ngraph::element::u64:
+    case ngraph::element::f16:
+    case ngraph::element::f32:
+        return true;
+    default:
+        break;
+    }
+    return false;
+}
+
+bool op::v8::MaxPool::evaluate(const HostTensorVector& outputs, const HostTensorVector& inputs) const {
+    NGRAPH_OP_SCOPE(v8_MaxPool_evaluate);
```

Comment:
Do we really need to have evaluate for this operation?

Reply:
Agree with @ilyachur. The fallback to ref implementation is supported on CPU only and MaxPoll-8 will be supported on CPU. So it doesn't make sense to have it in the evaluate.

Reply:
@ilyachur @lazarevevgeny does that mean I should move it to evaluates_map or do you mean something else?

Reply:
Yes, looks like at the current moment we don't need to have evaluate method for this operation, but this implementation has a lot of template code and can have significant impact on binary size.